### ISSUE TITLE
Set HTTP status code

### ISSUE THREAD

**Issue by zammitt** [Dasync/Dasync#5]
Is it possible to set the HTTP status code when an exception is thrown by a http-enabled service?

From what I can tell, dasync defaults to 400 when any exception is thrown.

**Comment by zammitt** [comment]
I'd like to be able to return 404 when a resource is not found and 500 when an unexpected exception is thrown.

Either a way to provide a map of exception type to status code or a way to decorate custom exceptions with the desired status code would likely work.

**Comment by kind-serge** [comment]
The mapping of an exception to a status code is doable.
Can you elaborate on the meaning of "unexpected exception"? How do you determine that?

**Comment by zammitt** [comment]
If we had a map of Exception Type -> Status Code then an "unexpected exception" would be any exception not found in the map.   These exceptions would ideally return a 500 Internal Server Error since it likely means something unexpected on the backend failed.

**Comment by kind-serge** [comment]
Hi @zammitt, the feature requested is available in v1.1.0:
https://www.nuget.org/packages/Dasync.AspNetCore/1.1.0

Here is sample usage:
```csharp
    public class Startup
    {
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            app.UseDasync(Sample.CommunicationModel);

            app.MapExceptionToHttpCode(
                (typeof(FileNotFoundException), 404),
                (typeof(Exception), 500));
        }
    }
```
It works like the `catch` statement, so any exception type derived from specified ones will have a respective status code.

**Comment by zammitt** [comment]
Awesome!  Thanks Serge! I will try it out as soon as we upgrade to this version.
